### PR TITLE
Create basic library structure

### PR DESCRIPTION
**Pull request by fraya**
Adapt library to include basic structure:

- Standard library structure of modules and directories.
- Package description.
- Basic tests.
- Basic documentation.

### CODE REVIEW

File: documentation/source/reference.rst [hunk 1]
```diff
@@ -0,0 +1,33 @@
+Reference
+=========
+
+.. current-library:: zlib
+.. current-module:: zlib		     
+
+Utility functions
+-----------------
+
+This functions are implemented on top of the basic zlib
```
Comment (cgay) [inline documentation/source/reference.rst]: These functions...

File: tests/zlib-test-suite.lid [hunk 2]
```diff
@@ -0,0 +1,4 @@
+Library: zlib-test-suite
+Files: library.dylan
+       zlib-test-suite
```
Comment (cgay) [inline tests/zlib-test-suite.lid:3]: Either omit all file extensions (.dylan) or use them everywhere.

**Review by cgay — CHANGES_REQUESTED** [review]
LGTM

File: documentation/source/reference.rst (hunk 1 above)
Comment (fraya) [reply documentation/source/reference.rst]: Fixed.

File: tests/zlib-test-suite.lid (hunk 2 above)
Comment (fraya) [reply tests/zlib-test-suite.lid:3]: Fixed.

**Pull request merged**